### DISCUSSION TITLE
How can I test my protected routes using the dev provider?

### DISCUSSION THREAD

**Discussion by jamesbalcombe83** [go-pkgz/auth#177] [Q&A]
I hope someone can help. I want to write some tests for my protected routes, so I assume I need to use the dev provider in some way, but it's not clear to me what the process would look like in the test.

I've added the `dev` provider to the auth service in my router, and added the code to run the `devAuthServer` in a go routine before my routes, but what request should I make inside of the test? I suppose that the authorize route needs to be called somehow?

**Comment by paskal** [comment]
Please see remark42 for example: construction of auth service [here](https://github.com/umputun/remark42/blob/c9b6f9272f5de893cdec49acdb04ffc9d610dc64/backend/app/cmd/server.go#L1139) and [here](https://github.com/umputun/remark42/blob/c9b6f9272f5de893cdec49acdb04ffc9d610dc64/backend/app/cmd/server.go#L851), [set up of the handlers](https://github.com/umputun/remark42/blob/master/backend/app/rest/api/rest.go#L226), [protecting routes with that auth module](https://github.com/umputun/remark42/blob/master/backend/app/rest/api/rest.go#L284-L289). [Here](https://remark42.com/demo/) is how it looks like in the frontend.
